newrelic/hooks/application_celery.py

```diff
@@ -87,16 +88,34 @@ def _application():
         #      running inside of an existing transaction, we want to create
         #      a new background transaction for it.
 
-        if transaction and (transaction.ignore_transaction or
-                transaction.stopped):
+        if transaction and (transaction.ignore_transaction or transaction.stopped):
             return wrapped(*args, **kwargs)
 
         elif transaction:
             with FunctionTrace(_name, source=instance):
                 return wrapped(*args, **kwargs)
 
         else:
-            with BackgroundTask(_application(), _name, 'Celery', source=instance):
+            with BackgroundTask(_application(), _name, "Celery", source=instance) as transaction:
+                # Attempt to grab distributed tracing headers
+                try:
+                    # Headers on earlier versions of Celery may end up as attributes
+                    # on the request context instead of as custom headers. Handler this
+                    # by defaulting to using vars() if headers is not available
+                    headers = getattr(wrapped.request, "headers", None) or vars(wrapped.request)
+
+                    settings = transaction.settings
+                    if headers is not None and settings is not None:
+                        if settings.distributed_tracing.enabled:
+                            transaction.accept_distributed_trace_headers(headers, transport_type="AMQP")
+                        elif transaction.settings.cross_application_tracer.enabled:
```

Comment:
I really wonder if this is worth supporting as we've deprecated this feature and plan on EOLing it soon but nbd either way.

Reply:
It's just copied from the MessageTransaction which I'm considering just using anyway.